radsat-sk/operation/message/protobuf/RProtobuf.c

```diff
@@ -37,20 +38,23 @@ uint8_t protoEncode(RadsatMessage* rawMessage, uint8_t* outgoingBuffer) {
 
 
 /**
- * Decode an encoded protobuf message, and confirm that the contents match the header.
+ * Decode an encoded protobuf message.
  *
- * @param incomingBuffer The buffer containing the encoded message/header.
+ * @param incomingBuffer The buffer containing the encoded message (no header).
  * @param decodedMessage The message that will be populated with the decoded message.
- * @return true if the message is decoded successfully, false otherwise.
+ * @return 0 if the message is decoded successfully, otherwise error occured.

```

Comment:
Your decoder has a successful return on 0 but your encoder has a failure return on 0. It would be nice to standardize this but not in this PR. I want to investigate error handling and will implement something then. I just want you to be aware of this for the furture